include/fiction/utils/truth_table_utils.hpp

```diff
@@ -409,6 +412,26 @@ namespace fiction
     return std::vector<kitty::dynamic_truth_table>{table1, table2};
 }
 
+/**
+ * This function evaluates the given multi-output truth table at the given input index.
+ *
+ * @param truth_tables The truth tables to evaluate.
+ * @param current_input_index The index representing the current input pattern.
+ * @return The output of the truth tables for the given input index.
+ */
+[[nodiscard]] inline uint64_t evaluate_output(const std::vector<kitty::dynamic_truth_table>& truth_tables,
+                                              const uint64_t current_input_index) noexcept
+{
+    assert(truth_tables.size() <= 64 && "Number of truth tables exceeds 64");
```

Comment:
An `assert` statement isn't appropriate error handling for a function that so heavily relies on proper input sizing.

Reply:
I'd still reconsider whether `uint64_t` is the best return type here. Why not `std::vector<bool>`?

Reply:
The reason is that we also use ``uint64_t`` for the ``input_index``.

Reply:
> An `assert` statement isn't appropriate error handling for a function that so heavily relies on proper input sizing.

Absolutely, but what can we do better? Throwing an exception can have a bad impact on runtime, right?

Reply:
> The reason is that we also use `uint64_t` for the `input_index`.

You're right. The input index cannot be larger than 64 bits. However, `truth_tables`, a dynamically sized vector, can be larger than 64 entries.

Reply:
> > An `assert` statement isn't appropriate error handling for a function that so heavily relies on proper input sizing.
> 
> Absolutely, but what can we do better, @marcelwa? Throwing an exception can have a bad impact on runtime, right?

That is also true. If we use this function a lot, an exception would indeed negatively impact the performance significantly... difficult situation...

Reply:
We could work with ``std::optional``, but that makes it difficult to use elsewhere in the code.

Reply:
Or we use ``std::array`` at set the size at compile-time, right?

Reply:
> Or we use `std::array` and set the size at compile-time, right?

In that case, you would **always** pass 64 truth tables, though...

Reply:
true

Reply:
I guess this is a situation where safety has to be sacrificed for the sake of performance.

Reply:
I guess so...